src/QirRuntime/test/Standalone/InputTypes/qir-driver.cpp

```diff
@@ -0,0 +1,145 @@
+        std::vector<int64_t> anIntegerArray;
+        CLI::Option *anIntegerArrayOpt = app.add_option(
+            "--anintegerarray",
+            anIntegerArray,
+            "An integer array");
+
+        anIntegerArrayOpt->required();
+
+        // TODO: Add option for Q# Tuple<T> type.
+
+        // With all the options added, parse arguments from the command line.
+        CLI11_PARSE(app, argc, argv);
+
+        // Redirect the simulator output from std::cout if the --simulator-output option is present.
+        std::ostream* simulatorOutputStream = &std::cout;
+        std::filebuf simulatorOutputFileBuffer;
+        if (!simulatorOutputFileOpt->empty()){
+            std::cout << "Simulator Output File: " << simulatorOutputFile << std::endl;
+            simulatorOutputFileBuffer.open(simulatorOutputFile, std::ios::out);
+            std::ostream simulatorOutputFileStream(&simulatorOutputFileBuffer);
+            // TODO: Call into the QIR runtime API to redirect the output of the simulator.
+            simulatorOutputStream = &simulatorOutputFileStream;
+        }
+
+        // TODO: This is for testing purposes only.
+        (*simulatorOutputStream) << "SIMULATOR OUTPUT:\nSample Output\n";
+
+        // Redirect the Q# opertion output from std::cout if the --operation-output option is present.
+        std::ostream* operationOutputStream = &std::cout;
+        std::filebuf operationOutputFileBuffer;
+        if (!operationOutputFileOpt->empty()){
+            std::cout << "Operation Output File: " << operationOutputFile << std::endl;
+            operationOutputFileBuffer.open(operationOutputFile, std::ios::out);
+            std::ostream operationOutputFileStream(&operationOutputFileBuffer);
+            operationOutputStream = &operationOutputFileStream;
+        }
+
+        // TODO: Remove this after the Message Q# function is integrated into the QIR runtime.
+        (*simulatorOutputStream) << anInt << std::endl;
+        (*simulatorOutputStream) << aDouble << std::endl;
+        (*simulatorOutputStream) << aBool << std::endl;
+        for (int64_t n : anIntegerArray) {
+            (*simulatorOutputStream) << n << " ";
+        }
+
+        (*simulatorOutputStream) << std::endl;
+
+        // Start simulation.
+        // TODO: Use the pattern suggested by Irina.
+        SetupQirToRunOnFullStateSimulator();
+        // TODO: Pass the parsed arguments to the entry-point operation.
+        Microsoft__Quantum__Testing__QIR__InputTypes__body(anInt, aDouble, 0, nullptr);
+        simulatorOutputStream->flush();
+        (*operationOutputStream) << "OPERATION OUTPUT:\n";
+        operationOutputStream->flush();
+
+        // Close opened file buffers;
+        if (!simulatorOutputFileOpt->empty()){
+            simulatorOutputFileBuffer.close();
+        }
+
+        if (!operationOutputFileOpt->empty()){
+            operationOutputFileBuffer.close();
+        }
+    } catch(...) {
+        return 1;
+    }
```

Comment:
It is typically easier to catch the unexpected exception bugs if the exceptions are _not caught_. In that case the app either immediately crashes upon the unexpected exception (if the debugger is not attached), or breaks in the debugger at the moment the exception is thrown (if the debugger is attached).
Currently your code catches all the exceptions, prints _nothing to console_, and just returns 1. This can make an impression that the test has run successfully (while it has terminated upon unexpected exception).
You may also find useful the macros of Catch2 test library, see `REQUIRE(..)`, `REQUIRE_THROWS_AS(..)` in the other tests and [here](https://github.com/catchorg/Catch2/blob/devel/docs/assertions.md#top).

Reply:
If we want this sample to look as close as possible to the code a user might produce and, thus, would rather avoid testing it with Catch2, we can still test the final binary through ctest (and we probably should).

Reply:
As Irina mentioned, the goal is to make this sample as close as possible to the use that will be produced automatically produced so we shouldn't use Catch2 in this case. I removed the global try-catch and will add a test as suggested by Irina in this [comment](https://github.com/microsoft/qsharp-runtime/pull/520#discussion_r581438121).